src/System.Private.CoreLib/src/System/Utf8String.cs

```diff
@@ -339,5 +341,16 @@ public override string ToString()
                 return retVal;
             }
         }
+
+        // TODO! Decide if string interning should be a public feature
```

Comment:
Probably should offer a data version if so (`Span<uchar>` or whatever the data type is) rather than always requiring `Utf8String` creation first to then swap with interned string

Reply:
Seems like a good plan. I'm going to push this to actually compiling and working on non-Windows, before I do so though, so it probably won't be in this PR (note that this PR is against a feature branch not against master)